### PR TITLE
Add mentee-side mentor profile page

### LINKED ISSUE #75: Mentee Side - Mentor page
#### Description
- each mentor should have their own page
- differentiate by having the mentor id in the url as a parameter
- for the videos, the big one is the one that has "pinned" as one of the tags. The other ones without "pinned" in its list of tags are the smaller videos. 

#### Screenshots
image

### PR DESCRIPTION
**Pull request by Feeshay**
## Status:
:rocket: Ready

## Description :star2:
* Refactor/clean up profile page code (move some logic to ProfileContent component)
* Refactor/clean up gallery code (move some logic to new MentorCard component)
* Add component/route for public profile page (PublicProfile)
* Add component for video display (ProfileVideos)
* CSS and stuff for public profile page

image

Resolves #75

### CODE REVIEW

File: frontend/src/components/ProfileVideos.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  const defaultVideo = <div className="video-default-preview"> </div>;
+
+  return (
+    <div>
+      <h1>
+        <b>Videos</b>
+      </h1>
+      <hr className="mentor-profile-videos-divider" />
+      {/* Temporary display to show search updating */}
```
Comment (AngelaLuo49021) [inline frontend/src/components/ProfileVideos.js:20]: Can you finish this component as a part of this issue?
- [ ] fix up videos dummy data in the database
- [ ] display videos (make sure to check for the "pinned" tag for the large video

**Review by AngelaLuo49021 — CHANGES_REQUESTED** [review]
Everything else looks gucci, frontend god or whateva

File: frontend/src/components/MentorCard.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        <h4>
+          <LinkOutlined style={styles.icon} />
+          <a className="gallery-links" href={props.website}>
+            {props.website}
+          </a>
+        </h4>
+        <h4>
+          <LinkedinOutlined style={styles.icon} />
+          <a className="gallery-links" href={props.linkedin}>
+            linkedin
```
Comment (chyku) [inline frontend/src/components/MentorCard.js:77]: is this supposed to be `linkedin` or like `props.linkedin` or something

File: frontend/src/components/ProfileVideos.js (hunk 1 above)
Comment (chyku) [reply frontend/src/components/ProfileVideos.js:20]: i wouldn't mind if this were put in another PR, this one is large enough

File: frontend/src/components/pages/Profile.js [hunk 3]
```diff
@@ -27,113 +20,13 @@ function Profile() {
...
-            <div className="mentor-profile-heading">
-              <b>Specializations</b>
-            </div>
-            <div>{getSpecializationTags(mentor.specializations || [])}</div>
-            <br />
-            <div className="mentor-profile-heading">
-              <b>Education</b>
-            </div>
-            <div>{getEducations(mentor.education)}</div>
+            <ProfileContent mentor={mentor} button={"mentor"} />
```
Comment (chyku) [inline frontend/src/components/pages/Profile.js:29]: would it be possible to make the prop "viewer" or something instead to make it more clear what it does

**Review by chyku — COMMENTED** [review]
some qs

File: frontend/src/components/MentorCard.js (hunk 2 above)
Comment (Feeshay) [reply frontend/src/components/MentorCard.js:77]: don't look at me I just copy/pasted
Right now it's just displaying a string that says "linkedin"; I think that's ok?
image

File: frontend/src/components/ProfileVideos.js (hunk 1 above)
Comment (Feeshay) [reply frontend/src/components/ProfileVideos.js:20]: I'll make another pr/issue for this

File: frontend/src/components/MentorCard.js (hunk 2 above)
Comment (chyku) [reply frontend/src/components/MentorCard.js:77]: LOL i think thats a bug can u fix it real quick

**Review by AngelaLuo49021 — APPROVED** [review]
gucci

**Review by chyku — APPROVED** [review]
looks good lmk if u fix that linkedin thing

**Pull request merged**